### ISSUE TITLE
FTP File Upload Resume

### ISSUE THREAD

**Issue by derekbking** [drakkan/sftpgo#666]
It seems that there is no support for FTP file upload resume. Is this feature suppose to be supported or are there any plans to support it?

When testing SFTP I noticed that resume is working.

Great work on this library btw, loving it so far!

**Comment by drakkan** [comment]
Hi,

the FTP `APPE` command is supported, can you please provide more info? Your ftp client and if you can a wireshark capture or the exact FTP commands issued. Thanks

**Comment by drakkan** [comment]
Hi,

I just tested using WinSCP as client, here are the FTP commands issued:

```
REST 402358272
350 OK
STOR <file name>
```

and the resume works fine. This is the restart procedure described [here](https://datatracker.ietf.org/doc/html/rfc3659#section-5.1).

As explained above SFTPGo support the `APPE` command too. 

Maybe your FTP client is trying to use another way to resume the upload, for example we don't support this (expired) [RFC draft](https://datatracker.ietf.org/doc/html/draft-bryan-ftp-range) but we don't include `RANG STREAM` in the `FEAT` response so the client should not try this method.

If you provide more info about your FTP client and I can reproduce the issue myself, I can investigate more, thanks

**Comment by derekbking** [comment]
Hi, I am testing with FileZilla.

I can reproduce the issue by uploading a large file. Then on FileZilla, right click the item in the queued files tab and unchecking and rechecking the processes queue option.
![image](https://user-images.githubusercontent.com/3951143/148692834-bac9057c-2a24-491a-8064-3efe56b1120c.png)
I am then prompted by FileZilla with this message 
![image](https://user-images.githubusercontent.com/3951143/148692780-479035b3-dce4-4f44-b840-7f51b357e558.png)

I select resume and the file upload starts from the beginning. I can provide more details if necessary.

**Comment by derekbking** [comment]
I just tested and it seems FileZilla does not resume ASCII based files on FTP. It also assumes files with no extension are ASCII based per default settings.
![image](https://user-images.githubusercontent.com/3951143/148693346-1fb10d42-d2b8-47ae-a19f-43048d15a112.png)

Renaming to file to .zip causes resume to work properly. I found some interesting discussion on [FileZilla's forums](https://forum.filezilla-project.org/viewtopic.php?t=33877). Seems to be related to line ending conversion.

**Comment by drakkan** [comment]
Hi,

I see, it works if you uncheck "Treat files without extension as ASCII file".  In ASCII mode upload resume is not possibile since the size for the file could be different from client and server for line ending conversion and so the resume could produce a corrupted file. 

Even if the FTP client sends a `REST` command and the transfer mode is ASCII, SFTPGo will reply with the 501 errore code and the message `Resuming transfers not allowed in ASCII mode`.

So basically if a client wants to resume an upload it must use the binary mode
